pkgs/development/tools/devpi-client/default.nix

```diff
@@ -1,40 +1,66 @@
 { stdenv
-, pythonPackages
+, buildPythonApplication
+, fetchPypi
+# buildInputs
 , glibcLocales
+, pkginfo
+, check-manifest
+# propagatedBuildInputs
+, py
+, devpi-common
+, pluggy
+, setuptools
+# CheckInputs
+, pytest
+, pytest-flake8
+, webtest
+, mock
 , devpi-server
+, tox
+, sphinx
+, wheel
 , git
 , mercurial
 } :
 
-pythonPackages.buildPythonApplication rec {
+buildPythonApplication rec {
   pname = "devpi-client";
-  version = "4.1.0";
+  version = "5.0.0";
 
-  src = pythonPackages.fetchPypi {
+  src = fetchPypi {
     inherit pname version;
-    sha256 = "0f5jkvxx9fl8v5vwbwmplqhjsdfgiib7j3zvn0zxd8krvi2s38fq";
+    sha256 = "0hyj3xc5c6658slk5wgcr9rh7hwi5r3hzxk1p6by61sqx5r38v3q";
   };
 
-  checkInputs = with pythonPackages; [
-                    pytest pytest-flakes webtest mock
-                    devpi-server tox
-                    sphinx wheel git mercurial detox
-                    setuptools
-                    ];
+  buildInputs = [ glibcLocales pkginfo check-manifest ];
+
+  propagatedBuildInputs = [ py devpi-common pluggy setuptools ];
+
+  checkInputs = [
+    pytest pytest-flake8 webtest mock
+    devpi-server tox
+    sphinx wheel git mercurial
+    setuptools
```

Comment:
since setuptools is in propagatedBuildInputs, you should (in theory), be able to remove this.